tools/ci/annotate_dm.py

```diff
@@ -0,0 +1,31 @@
+import sys
+import re
+import os.path as path
+
+# Usage: python3 annotate_dm.py [filename]
+# If filename is not provided, stdin is checked instead
+
+raw_output = ""
+
+if len(sys.argv) > 1:
+    if not path.exists(sys.argv[1]):
+        print(f"Annotations file '{sys.argv[1]}' does not exist")
+        sys.exit(1)
+    with open(sys.argv[1], 'r') as f:
+        raw_output = f.read()
+elif not sys.stdin.isatty():
+    raw_output = sys.stdin.read()
+
+# Remove ANSI escape codes
+raw_output = re.sub(r'(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]', '', raw_output)
+
+annotation_regex = r'(?P<filename>.*?), line (?P<line>\d+), column (?P<column>\d+):\s{1,2}(?P<type>error|warning): (?P<message>.*)'
+
+has_issues = False
+
+for annotation in re.finditer(annotation_regex, raw_output):
+    print(f"::{annotation['type']} file={annotation['filename']},line={annotation['line']},col={annotation['column']}::{annotation['message']}")
+    has_issues = True
+
+if not has_issues:
+    print("No DM issues found")
```

Comment:
```suggestion
if not has_issues:
    print("No DM issues found")
    return
    
sys.exit(1)
```

I'm 99% sure the old behavior was that the action would handle failing CI if we had issues via DreamChecker so that behavior should be preserved. let me know if i'm wrong

Reply:
[It didn't fail CI.](https://github.com/yogstation13/DreamAnnotate/blob/master/src/index.ts) What caused CI to fail was the exit code 1 from dreamchecker, the annotate step would always run regardless if the previous step failed. And [since runners enforce `set -o pipefail`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference), the runner would ignore the exit code coming from `annotate_dm.sh` anyways.

Reply:
alright good to know